scarb/tests/fmt.rs

```diff
@@ -270,3 +303,77 @@ fn workspace_with_root() {
     let content = fs::read_to_string(t.child("second/src/lib.cairo")).unwrap();
     assert_eq!(content, SIMPLE_FORMATTED);
 }
+
+#[test]
+fn workspace_emit_with_root() {
+    let t = TempDir::new().unwrap().child("test_workspace");
+    let pkg1 = t.child("first");
+    ProjectBuilder::start()
+        .name("first")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .build(&pkg1);
+    let pkg2 = t.child("second");
+    ProjectBuilder::start()
+        .name("second")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .dep("first", &pkg1)
+        .build(&pkg2);
+    let root = ProjectBuilder::start()
+        .name("some_root")
+        .lib_cairo(SIMPLE_ORIGINAL)
+        .dep("first", &pkg1)
+        .dep("second", &pkg2);
+    WorkspaceBuilder::start()
+        .add_member("first")
+        .add_member("second")
+        .package(root)
+        .build(&t);
+
+    Scarb::quick_snapbox()
+        .arg("fmt")
+        .arg("--emit")
+        .arg("stdout")
+        .current_dir(&t)
+        .assert()
+        .failure()
+        .stdout_matches(indoc! {"\
+            fn main() -> felt252 {
+                42
+            }
+
+        "});
+
+    let content = fs::read_to_string(t.child("src/lib.cairo")).unwrap();
+    assert_eq!(content, SIMPLE_ORIGINAL);
+    let content = fs::read_to_string(t.child("first/src/lib.cairo")).unwrap();
+    assert_eq!(content, SIMPLE_ORIGINAL);
+    let content = fs::read_to_string(t.child("second/src/lib.cairo")).unwrap();
+    assert_eq!(content, SIMPLE_ORIGINAL);
+
+    Scarb::quick_snapbox()
+        .args(["fmt", "--workspace", "--emit", "stdout"])
```

Comment:
added a test for workspace,
it's unclear to me what behaviour the stdout should have for formatting multiple files at once,
i will search what the linters expect, maybe that some separator symbol and file path be introduced?

Reply:
I would prefer if we followed what Cargo does in this regard (i.e. `cargo fmt -- --emit stdout`), as Scarb is mostly inspired by Cargo's approach. This means printing first full path to file, terminated with `:`, then from the next line file contents. 
I have not looked into tools that may use this format - please do! 🙇

Reply:
Great idea! I did that